luaui/barwidgets.lua

```diff
@@ -1189,6 +1189,12 @@ end
 
 
 function widgetHandler:Update()
+	
+	if collectgarbage("count") > 1200000 then 
```

Comment:
why 1.2GB specifically?

Reply:
You could also consider increasing the maximum. That would reduce the overhead of GC as well (apparently it runs proportionally to usage %)

Reply:
Our GC rate is pretty fine tuned, though I will likely tune it lower. The problem is a slight runaway effect when too much mem is allocated. In detail:

We currently allow 1ms of time for GC pass. The probability of GC happening is (alloced/max). Thus if there is too much allocation, and also a lot of garbage being created, then 1ms of time isnt enough to collect enough garbage to stop growing. 

This is literally an emergency collector (outside of that engine bug under fixing), no bar-shipped code should ever trigger this. User widgets are a different matter, in that case its a good safety net.

Reply:
`though I will likely tune it lower` specifically during catch-up phase, synced GC kicks in too often, there its 10-20% of CPU load.